src/app/sessions/services/sessions-statistics-service.ts

```diff
@@ -0,0 +1,27 @@
+import { Injectable } from '@angular/core';
+import { schema } from 'normalizr';
+import * as moment from 'moment';
+
+@Injectable()
+export class SessionsStatisticsService {
```

Comment:
I can see here OO problem because this service right now acts more like a procedural than object-oriented code.
There is a notion that all methods here are time-related calculations. If we go this path, 2 of them are session independent and their naming could be slightly different I personally feel that changing method names to end with 'ity' (ex. calculateAvailability) or adding 'Time' suffix (ex. calculateAvailableTime) is better. With that in mind, method calculateAllocated should focus on calculating time with a generic approach, changing session from the parameters and taking an array of times or objects/types with time field - not sure if later is doable with typescript) and method calling it will need to provide needed data.
Having those changes the name of the class would have to change accordingly because even though we know it's sessions statistics, we have here only time-related calculations.

Reply:
I thought of this service to be suited for our session. If we make it generic then those methods will become trivial one-liners and I don't see a value in that. I don't think we could use it anywhere else.